examples/extensions/commands/cmd_clean.py

```diff
@@ -0,0 +1,53 @@
+from conan.api.conan_api import ConanAPIV2
+from conans.cli.command import conan_command, OnceArgument
+from conans.cli.output import Color, ConanOutput
+from conans.client.userio import UserInput
+
+info_color = Color.BRIGHT_WHITE
+recipe_color = Color.BRIGHT_BLUE
+removed_color = Color.BRIGHT_YELLOW
+
+
+@conan_command(group="Custom commands")
+def clean(conan_api: ConanAPIV2, parser, *args):
+    """
+    Clean the earlier recipe revisions (and their packages) and the package ones from local or remote.
+
+    Important! Keeps the LATEST package revisions belonging to the LATEST recipe revision.
+    """
+    parser.add_argument('-r', '--remote', action=OnceArgument,
+                        help='Will remove from the specified remote')
+    parser.add_argument('--force', default=False, action='store_true',
+                        help='Remove without requesting a confirmation')
+    args = parser.parse_args(*args)
+
+    def confirmation(message):
+        return args.force or ui.request_boolean(message)
+
+    out = ConanOutput()
+    ui = UserInput(conan_api.config.get("core:non_interactive"))
+    remote = conan_api.remotes.get(args.remote) if args.remote else None
+    output_remote = remote or "Local cache"
+    # Getting all the recipes
+    recipes = conan_api.search.recipes("*/*", remote=remote)
+    for recipe in recipes:
+        out.writeln(f"{str(recipe)}", fg=recipe_color)
+        # All RREVS and latest one
+        all_rrevs = conan_api.search.recipe_revisions(f"{str(recipe)}#*", remote=remote)
```

Comment:
I think we don't want to use ``search.recipe_revisions()`` but ``list.recipe_revisions()``

Reply:
Why? The `conan remove` command is using the `search` one internally, so why should it be different?

Reply:
The ``search`` api is intended to be used with patterns, by users, for ``upload`` and ``remove`` commands. But in this case, you are already using full, deterministic recipe references that you obtained with the first call to the search api, so you can call directly the ``list`` api.